solls/LanguageServer.h

```diff
@@ -0,0 +1,117 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+#pragma once
+
+#include <solls/FileReader.h>
+
+#include <liblsp/Server.h>
+#include <liblsp/VFS.h>
+#include <liblsp/protocol.h>
+
+#include <libsolidity/interface/CompilerStack.h>
+#include <libsolidity/interface/ReadFile.h>
+
+#include <json/value.h>
+
+#include <boost/filesystem/path.hpp>
+
+#include <functional>
+#include <map>
+#include <memory>
+#include <optional>
+#include <ostream>
+#include <string>
+#include <string_view>
+#include <unordered_map>
+#include <variant>
+
+namespace solidity {
```

Comment:
Should it really be in the `solidity` namespace instead of `solidity::frontend` or something else?

Reply:
It was `lsp` and I mentioned `solidity` but probably wasn't clear I meant `solidity::lsp`.